ci-operator/step-registry/openstack/provision/etcd-disk-speed/openstack-provision-etcd-disk-speed-commands.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export OS_CLIENT_CONFIG_FILE="${SHARED_DIR}/clouds.yaml"
+
+function info() {
+	printf '%s: %s\n' "$(date --utc +%Y-%m-%dT%H:%M:%SZ)" "$*"
+}
+
+function wait_for_etcd_patch_done() {
+  info 'INFO: Waiting for patch etcd cluster...'
+  if
+    ! oc wait --timeout=30m --for=jsonpath='{.spec.controlPlaneHardwareSpeed}'="Slower" etcd cluster 1>/dev/null || \
+    ! oc wait --timeout=30m --for=jsonpath='{.status.conditions[0].status}'="False" co etcd 1>/dev/null || \
+    ! oc wait --timeout=30m --for=jsonpath='{.status.conditions[1].status}'="False" co etcd 1>/dev/null || \
+    ! oc wait --timeout=30m --for=jsonpath='{.status.conditions[2].status}'="True" co etcd 1>/dev/null; then
+      info "ERROR: Etcd patch failed..."
```

Comment:
Waiting for conditions by their list position isn't ideal. Another way is to run:

```bash
for i in {1..30}; do
  ETCD_CO_AVAILABLE=$(oc get co etcd | grep etcd | awk '{print $3}')
  if [[ "${ETCD_CO_AVAILABLE}" == "True" ]]; then
    echo "Patched successfully!"
    break
  fi
  sleep 15
done
if [[ "${ETCD_CO_AVAILABLE}" != "True" ]]; then
  echo "Etcd patch failed..."
  exit 1
fi
```